Demo/Demo/ViewModels/BaseViewModel.swift

```diff
@@ -27,39 +27,32 @@ class BaseViewModel: ObservableObject {
     }
 
     func updateOrderID(with orderID: String) {
-        DispatchQueue.main.async {
-            self.orderID = orderID
-        }
+        self.orderID = orderID
     }
 
-    func createOrder(amount: String?, completion: @escaping (String?) -> Void = { _ in }) {
+    func createOrder(amount: String?) async -> String? {
         updateTitle("Creating order...")
-
-        guard let amount = amount else { return }
+        guard let amount = amount else { return nil }
 
         let amountRequest = Amount(currencyCode: "USD", value: amount)
         let orderRequestParams = CreateOrderParams(
             intent: DemoSettings.intent.rawValue.uppercased(),
             purchaseUnits: [PurchaseUnit(amount: amountRequest)]
         )
 
-        DemoMerchantAPI.sharedService.createOrder(orderParams: orderRequestParams) { result in
-            switch result {
-            case .success(let order):
-                self.updateTitle("Order ID: \(order.id)")
-                self.updateOrderID(with: order.id)
-                print("✅ fetched orderID: \(order.id) with status: \(order.status)")
-            case .failure(let error):
-                self.updateTitle("Your order has failed, please try again")
-                print("❌ failed to fetch orderID: \(error)")
-            }
-            DispatchQueue.main.async {
-                completion(self.orderID)
-            }
+        do {
+            let order = try await DemoMerchantAPI.sharedService.createOrder(orderParams: orderRequestParams)
```

Comment:
Curious about the change in error handling behavior here - Is do/catch the convention with async/await? Could we instead return a tuple with order, or error in the await block so a merchant doesn't have to catch errors?

Reply:
Also realized this is demo code - but meant to call this out about the CardClient.approveOrder error handling block below (line 60-75)

Reply:
@sarahkoop that seems to be the convention which originated in JS i think (https://javascript.info/async-await). It is the same case for `android` and `coroutines`

Reply:
We could do a tuple but most of the examples I've seen have the catch block. I think it's similar with coroutines on the Kotling side as well.

One thing it does allow for is bubbling up exceptions. Like for BT when we have to "propagate" exceptions via callback, with `throw` they'll naturally propagate to the initial call site.

Reply:
For the record, [URLSession data(for:delegate:)](https://developer.apple.com/documentation/foundation/urlsession/3767352-data) does return a tuple, but the original `completionHandler` variant has three params `(Data?, URLResponse?, Error?)`. The new `async/await` variant will throw when it fails.

Reply:
Do catch is a pretty common pattern, so it makes sense for us to use that here